internal/cli/roles.go

```diff
@@ -229,46 +230,46 @@ func updateRoleCmd(cli *cli) *cobra.Command {
   auth0 roles update <role-id> -n myrole -d "awesome role" --json`,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) == 0 {
-				err := roleID.Pick(cmd, &inputs.ID, cli.rolePickerOptions)
-				if err != nil {
+				if err := roleID.Pick(cmd, &inputs.ID, cli.rolePickerOptions); err != nil {
 					return err
 				}
 			} else {
 				inputs.ID = args[0]
 			}
 
-			// Prompt for role name
-			if err := roleName.AskU(cmd, &inputs.Name, nil); err != nil {
+			var currentRole *management.Role
+			if err := ansi.Waiting(func() (err error) {
+				currentRole, err = cli.api.Role.Read(cmd.Context(), inputs.ID)
 				return err
+			}); err != nil {
+				return fmt.Errorf("failed to find role with ID %q: %v", inputs.ID, err)
 			}
 
-			// Prompt for role description
-			if err := roleDescription.AskU(cmd, &inputs.Description, nil); err != nil {
+			if err := roleName.AskU(cmd, &inputs.Name, currentRole.Name); err != nil {
 				return err
 			}
 
-			// Start with an empty role object. We'll conditionally
-			// hydrate it based on the provided parameters since
-			// we'll do PATCH semantics.
-			r := &management.Role{}
+			if err := roleDescription.AskU(cmd, &inputs.Description, currentRole.Description); err != nil {
+				return err
+			}
+
+			updatedRole := &management.Role{}
 
 			if inputs.Name != "" {
-				r.Name = &inputs.Name
+				updatedRole.Name = &inputs.Name
 			}
-
 			if inputs.Description != "" {
-				r.Description = &inputs.Description
+				updatedRole.Description = &inputs.Description
 			}
 
```

Comment:
Is it possible to provide no name _and_ no description on update? If so, might we detect that and prevent an API request that will inevitably fail?

Reply:
The API doesn't fail if we do not provide any name or description, it will simply behave as a GET.